lib/Doctrine/DBAL/Migrations/LazySchemaManipulator.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the LGPL. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\DBAL\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use ProxyManager\Factory\LazyLoadingGhostFactory;
+use ProxyManager\Proxy\LazyLoadingInterface;
+
+class LazySchemaManipulator
+{
+    /** @var  LazyLoadingGhostFactory */
+    private $proxyFactory;
+
+    /** @var SchemaManipulator */
+    private $originalSchemaManipulator;
+
+    public function __construct(LazyLoadingGhostFactory $proxyFactory, SchemaManipulator $originalSchemaManipulator)
+    {
+        $this->proxyFactory = $proxyFactory;
+        $this->originalSchemaManipulator = $originalSchemaManipulator;
+    }
+
+    /**
+     * @return Schema
+     */
+    public function createFromSchema()
+    {
+        return $this->proxyFactory->createProxy(
+            Schema::class,
+            function (& $wrappedObject, $method, array $parameters, & $initializer) {
+                $initializer   = null;
+                $wrappedObject = $this->originalSchemaManipulator->createFromSchema();
+
+                return true;
+            }
+        );
+    }
+
+    /**
+     * @param Schema $fromSchema
+     * @return Schema
+     */
+    public function createToSchema(Schema $fromSchema)
+    {
+        if ($fromSchema instanceof LazyLoadingInterface && ! $fromSchema->isProxyInitialized()) {
+            return $this->createFromSchema();
```

Comment:
this looks wrong to me. This would then read the database schema twice (for from and to schemas), while it should be done only once (it is a costly process)